File: src/main/java/ar/edu/unq/epers/unidad4/exception/GremioSinEspacioException.java
package ar.edu.unq.epers.unidad4.exception;

import ar.edu.unq.epers.unidad4.model.Gremio;

final public class GremioSinEspacioException extends RuntimeException {

    private static final long serialVersionUID = 1L;
    private final Gremio gremio;

    public GremioSinEspacioException(Gremio gremio) {
        this.gremio = gremio;
    }

    @Override
    public String getMessage() {
        return "El gremio [" + gremio.getNombre() + "] no tiene espacio para nuevos integrantes";
    }

}

File: src/main/kotlin/ar/edu/unq/epers/unidad4/exception/GremioSinEspacioException.kt
package ar.edu.unq.epers.unidad4.exception

import ar.edu.unq.epers.unidad4.model.Gremio

class GremioSinEspacioException(private val gremio: Gremio) : RuntimeException() {

    override val message: String?
    get() = "El gremio [${gremio.nombre}] no tiene espacio para nuevos integrantes"

    companion object {

        private val serialVersionUID = 1L
    }

}